### PR TITLE
Better logging and notification of setting on fire

### PR DESCRIPTION
**Pull request by optimumtact**
This also logs bump fire to the gamelog incase someone ever needs to
find out who set who on fire with bumping (left out of admin
notification) as it would be very spammy

### CODE REVIEW

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -73,6 +73,8 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 /obj/item/weapon/match/attack(mob/living/carbon/M, mob/living/carbon/user)
 	if(!isliving(M))
 		return
+	message_admins("[key_name_admin(user)] set (key_name_admin(M)) on fire")
+	log_game("[key_name(user)] set [key_name_user(M)] on fire")
```
Comment (Jordie0608) [inline code/game/objects/items/weapons/cigs_lighters.dm]: `[key_name(M)]`

File: code/game/objects/items/weapons/tools.dm [hunk 2]
```diff
@@ -319,6 +319,8 @@
 
 		if(isliving(O))
 			var/mob/living/L = O
+			message_admins("[key_name_admin(user)] set (key_name_admin(L)) on fire")
+			log_game("[key_name(user)] set [key_name(L)] on fire")
```
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: the mob isn't necessarily set on fire. It needs to have flammable fluid on them to be set on fire...

Comment (optimumtact) [inline code/game/objects/items/weapons/tools.dm]: Hmm that's a good point

Comment (KorPhaeron) [inline code/game/objects/items/weapons/tools.dm]: Check if they have firestacks

**Comment by optimumtact** [comment]
I've refactored IgniteMob to return true when it has set the user on fire and taken advantage of this to only log and notify when the mob is actually truly set on fire

@tgstation/commit-access

**Pull request merged**